pyo3-derive-backend/src/pyclass.rs

```diff
@@ -267,7 +267,7 @@ fn impl_class(
             quote! {
                 impl pyo3::freelist::PyClassWithFreeList for #cls {
                     #[inline]
-                    fn get_free_list() -> &'static mut pyo3::freelist::FreeList<*mut pyo3::ffi::PyObject> {
```

Comment:
I already merged this PR, but now I think about it should this actually return just `&mut Freelist`, so that it's not possible to get a reference that outlives the GIL lifetime?

Reply:
Hmm, yes, probably.

Reply:
👍 I'm going to write a quick follow-up PR to clean this up.